tools/AzureRM/AzureRM.psm1

```diff
@@ -75,29 +73,90 @@ function Update-AzureRM
   param(
   [Parameter(Position=0, Mandatory = $false)]
   [string]
-  $Repository,
+  $MajorVersion = $AzureMajorVersion,
   [Parameter(Position=1, Mandatory = $false)]
+  [string]
+  $Repository = "PSGallery",
+  [Parameter(Position=2, Mandatory = $false)]
   [ValidateSet("CurrentUser","AllUsers")]
   [string]
-  $Scope)
+  $Scope = "AllUsers")
 
-  if ([string]::IsNullOrWhiteSpace($Scope))
-  {
-    $Scope = "AllUsers"
-  }
-
-  Validate-AdminRights $Scope
+  Test-AdminRights $Scope
 
   Write-Output "Installing AzureRM modules."
 
-  Import-ModuleWithVersionCheck "AzureRM.Profile" $AzureRMProfileVersion $Repository $Scope
+  Install-ModuleWithVersionCheck "AzureRM.Profile" $MajorVersion $Repository $Scope
+
```

Comment:
I wonder if we'd do better with explicitly installing AzureRM.Resources and Azure.Storage before installing everything else